### ISSUE TITLE
[Event Request] - Codeunit Auto Create Sales Orders -  OnBeforeCreateNAVSalesOrdersFromSubmittedCRMSalesorders

### ISSUE THREAD

**Issue by DanielHaglo** [microsoft/ALAppExtensions#12615]
![image](https://user-images.githubusercontent.com/35260264/120293951-0352c000-c2c6-11eb-9f11-d9000d642d2a.png)

Can you create this event?

**Comment by Oleg-Romashkov** [comment]
Please always put the full detailed signature of the requested events and the parameters passed in the call.
You help us implementing the requests faster when your request contains AL code copied from VSCode instead of screenshots.

**Comment by DanielHaglo** [comment]
Hi Oleg,

My bad. Let me try again.

local procedure CreateNAVSalesOrdersFromSubmittedCRMSalesorders()
    var
        CRMSalesorder: Record "CRM Salesorder";
        IsHandled:Boolean;
    begin
        //Event
        OnBeforeCreateNAVSalesOrdersFromSubmittedCRMSalesorders(CRMSalesorder,IsHandled);
        if IsHandled then
            exit;
        //Event
        CRMSalesorder.SetRange(StateCode, CRMSalesorder.StateCode::Submitted);
        CRMSalesorder.SetFilter(LastBackofficeSubmit, '%1|%2', 0D, DMY2Date(1, 1, 1900));
        OnCreateNAVSalesOrdersFromSubmittedCRMSalesordersOnAfterCRMSalesorderSetFilters(CRMSalesorder);
        if CRMSalesorder.FindSet(true) then
            repeat
                Session.LogMessage('0000DET', StrSubstNo(StartingToCreateSalesOrderTelemetryMsg, CRMProductName.CDSServiceName(), CRMSalesorder.SalesOrderId), Verbosity::Normal, DataClassification::SystemMetadata, TelemetryScope::ExtensionPublisher, 'Category', CrmTelemetryCategoryTok);
                if CODEUNIT.Run(CODEUNIT::"CRM Sales Order to Sales Order", CRMSalesorder) then begin
                    Session.LogMessage('0000DEU', StrSubstNo(CommittingAfterCreateSalesOrderTelemetryMsg, CRMProductName.CDSServiceName(), CRMSalesorder.SalesOrderId), Verbosity::Normal, DataClassification::SystemMetadata, TelemetryScope::ExtensionPublisher, 'Category', CrmTelemetryCategoryTok);
                    Commit();
                end;
            until CRMSalesorder.Next() = 0;
    end;

//---- Event ----//

 [IntegrationEvent(false, false)]
    local procedure OnBeforeCreateNAVSalesOrdersFromSubmittedCRMSalesorders(var CRMSalesorder: Record "CRM Salesorder"; var IsHandled:Boolean)
    begin
    end;

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.
